src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Models/DependenciesViewModelFactory.cs

```diff
@@ -44,7 +44,7 @@ public ImageMoniker GetDependenciesRootIcon(bool hasUnresolvedDependencies)
 
         private IProjectDependenciesSubTreeProvider GetProvider(string providerType)
         {
-            return SubTreeProviders.FirstOrDefault(x => x.Value.ProviderType.Equals(providerType)).Value;
+            return SubTreeProviders.First(x => x.Value.ProviderType.Equals(providerType)).Value;
```

Comment:
This method's so simple that either exception would be meaningful in reality.

I generally favour the non-NRE when possible. My preference comes from trying to guess what threw NRE while looking at a stack trace from a user/customer.

My understanding is the info about what was null is only available in the debugger, not in a stack trace. 

Doesn't `First` say something like "sequence contained no elements"? It will also have `First` at the top of the stack trace, so it's clear it came from a call to `First`. By contrast a NRE can happen almost anywhere, making debugging harder.

Reply:
> My understanding is the info about what was null is only available in the debugger, not in a stack trace.

I'm sure you're right, I have no idea.

My preference is to use *OrDefault and check for null and log/throw a meaningful message, but my experience in this area is mostly from ORMs that tend to have lots of "sequences" in them

Reply:
> I'm sure you're right

Don't be :) I assume the exception type wasn't extended and that this is a debugger feature.